crates/bevy_render/src/pipelined_rendering.rs

```diff
@@ -17,13 +17,34 @@ use crate::RenderApp;
 #[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, AppLabel)]
 pub struct RenderExtractApp;
 
-/// Channel to send the render app from the main thread to the rendering thread
+/// Channels used by the main app to send and receive the render app.
 #[derive(Resource)]
-pub struct MainToRenderAppSender(pub Sender<SubApp>);
+pub struct RenderAppChannels {
```

Comment:
This is a public type, and will break anything reliant on it from the rename. This should be documented in the changelog and the migration guide.

This also can no longer be constructed outside of bevy_render, and I feel like exposing that might actually be an easy way to cause unusual deadlocks or crashes.

Might actually just be better to make this private since it's sort of an implementation detail of the plugin.

Reply:
I think I've seen people replace the update_rendering system with their own to be able to skip rendering sometimes. I think it was for a flutter integration. Making this private would make that a lot harder. I'll add a constructor instead.